app/models/vm_resize_request.rb

```diff
@@ -0,0 +1,26 @@
+class VmResizeRequest < MiqRequest
+  TASK_DESCRIPTION  = 'VM Resize'.freeze
+  SOURCE_CLASS_NAME = 'Vm'.freeze
+  ACTIVE_STATES     = %w(resized) + base_class::ACTIVE_STATES
+
+  validates :request_state, :inclusion => { :in      => %w(pending finished) + ACTIVE_STATES,
+                                            :message => "should be pending, #{ACTIVE_STATES.join(", ")} or finished"}
+  validate  :must_have_user
+
+  def self.make_request(request, values, requester, auto_approve = false)
+    values[:request_type] = :vm_resize
```

Comment:
`:request_type` should be `vm_cloud_reconfigure`.  This will be used in automate when processing the request events.